### PR TITLE
Base the rook ceph operator on Nautilus

### PR DESCRIPTION
**Pull request by travisn**
Signed-off-by: travisn <tnielsen@redhat.com>

**Description of your changes:**
With the v1.0 release we will plan on supporting nautilus. With that in mind, we should start running some integration tests on nautilus. This change moves the SmokeSuite over to nautilus, while also adding a test for starting the ceph nfs daemons with the new crds.

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

File: images/ceph/Makefile [hunk 1]
```diff
@@ -19,8 +19,10 @@ include ../image.mk
 
 CEPH_IMAGE = $(BUILD_REGISTRY)/ceph-$(GOARCH)
 
-CEPH_VERSION = v13.2.2-20181023
-BASEIMAGE = ceph/ceph-$(GOARCH):$(CEPH_VERSION)
+# FIX: Set the Nautilus image after it becomes available insetad of using ceph/daemon-base
+#CEPH_VERSION = v14.1.0-20180226
+#BASEIMAGE = ceph/ceph-$(GOARCH):$(CEPH_VERSION)
```
Comment (leseb) [inline images/ceph/Makefile]: You can now use `v14.2.0-20190319`.

**Comment by travisn** [comment]
Now with Nautilus as the base image, the integration tests are failing where Luminous is the target version. The operator is creating a block image for a PVC, then fails when it tries to retrieve the block image. The `rbd create` command succeeds, but the `rbd info` call fails with `exit status 95`. When I open the toolbox, I am able to list the image just fine:
```
# rbd ls replicapool
pvc-032e5745-4b38-11e9-8154-080027b3ec48
```

But if I try the command `rbd info replicapool/pvc-032e5745-4b38-11e9-8154-080027b3ec48` I see the same error. @dillaman are there known issues with rbd tools in nautilus when targeting luminous?

```
2019-03-20 17:47:48.062625 I | exec: Running command: rbd create replicapool/pvc-032e5745-4b38-11e9-8154-080027b3ec48 --size 20480 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring
I0320 17:47:48.064758       6 event.go:221] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"mysql-pv-claim", UID:"032e5745-4b38-11e9-8154-080027b3ec48", APIVersion:"v1", ResourceVersion:"3112", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "default/mysql-pv-claim"
2019-03-20 17:47:48.133798 W | cephclient: Requested image pvc-032e5745-4b38-11e9-8154-080027b3ec48 exists in pool replicapool. Continuing
2019-03-20 17:47:48.134127 I | exec: Running command: rbd info replicapool/pvc-032e5745-4b38-11e9-8154-080027b3ec48 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json
W0320 17:47:48.205158       6 controller.go:685] Retrying syncing claim "default/mysql-pv-claim" because failures 3 < threshold 15
E0320 17:47:48.205640       6 controller.go:700] error syncing claim "default/mysql-pv-claim": failed to provision volume with StorageClass "rook-ceph-block": Failed to create rook block image replicapool/pvc-032e5745-4b38-11e9-8154-080027b3ec48: failed to get image pvc-032e5745-4b38-11e9-8154-080027b3ec48 info after successfully creating it: failed to get image replicapool/pvc-032e5745-4b38-11e9-8154-080027b3ec48 info: Failed to complete '': exit status 95. 2019-03-20 17:47:48.198 7f4a3affd700 -1 librbd::image::RefreshRequest: failed to retrieve group: (95) Operation not supported
2019-03-20 17:47:48.198 7f4a3a7fc700 -1 librbd::image::OpenRequest: failed to refresh image: (95) Operation not supported
2019-03-20 17:47:48.198 7f4a3affd700 -1 librbd::image::RefreshRequest: failed to retrieve group: (95) Operation not supported
2019-03-20 17:47:48.198 7f4a3a7fc700 -1 librbd::image::OpenRequest: failed to refresh image: (95) Operation not supported
2019-03-20 17:47:48.198 7f4a3a7fc700 -1 librbd::ImageState: 0x5593fc10bfe0 failed to open image: (95) Operation not supported
2019-03-20 17:47:48.198 7f4a3a7fc700 -1 librbd::ImageState: 0x5593fc10bfe0 failed to open image: (95) Operation not supported
rbd: error opening image pvc-032e5745-4b38-11e9-8154-080027b3ec48: (95) Operation not supported
I0320 17:47:48.205499       6 event.go:221] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"mysql-pv-claim", UID:"032e5745-4b38-11e9-8154-080027b3ec48", APIVersion:"v1", ResourceVersion:"3112", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "rook-ceph-block": Failed to create rook block image replicapool/pvc-032e5745-4b38-11e9-8154-080027b3ec48: failed to get image pvc-032e5745-4b38-11e9-8154-080027b3ec48 info after successfully creating it: failed to get image replicapool/pvc-032e5745-4b38-11e9-8154-080027b3ec48 info: Failed to complete '': exit status 95. 2019-03-20 17:47:48.198 7f4a3affd700 -1 librbd::image::RefreshRequest: failed to retrieve group: (95) Operation not supported
```

**Comment by dillaman** [comment]
@travisn This is a known issue now :wink:. I'll open a PR to fix it ASAP, but it will take until the next Nautilus release to resolve.

**Comment by travisn** [comment]
@dillaman thanks! Is there another command that might work in place of `rbd info` to verify the existence of the image? Or could we just rely on the success of the `rbd create` command? If that's sufficient, we shouldn't need to call `rbd info`.

**Comment by dillaman** [comment]
@travisn If ```rbd create``` returns success, you can be sure that the image was properly created.

**Comment by travisn** [comment]
I'm testing now without the `rbd info` call. We were also retrieving the `size`, but it was also available in the original request so I think we should be fine.

**Comment by dillaman** [comment]
@travisn The fix is included here [1] -- you might also want to test deleting images against a Luminous cluster since that might not work either.

[1] https://github.com/ceph/ceph/pull/27080

**Comment by travisn** [comment]
@dillaman Yes, I do see it failing on rbd image deletion as well with the same error. We should be fine to wait for the fix though. This error won't block the rook CI, and by the time we release Rook 1.0 (~3 weeks) hopefully 14.2.1 would be out by then.

**Comment by travisn** [comment]
@leseb @galexrt could you take a final look at this? Assuming the CI passes now I'd like to get it in today.

File: tests/integration/base_file_test.go [hunk 2]
```diff
@@ -43,11 +43,23 @@ func runFileE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite.S
...
 	cleanupFilesystemConsumer(helper, k8sh, s, namespace, filesystemName, filePodName)
 	cleanupFilesystem(helper, k8sh, s, namespace, filesystemName)
 }
 
+func createNFSDaemons(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite.Suite, namespace string, filesystemName string) {
+	name := "my-nfs"
+	err := helper.NFSClient.Create(namespace, name, filesystemName+"-data0", 2)
+	require.Nil(s.T(), err)
+
+	err = helper.NFSClient.Delete(namespace, name)
```
Comment (galexrt) [inline tests/integration/base_file_test.go]: Function is called `createNFSDaemons`, but it creates and deletes the NFS Daemon?

Comment (travisn) [reply tests/integration/base_file_test.go]: I've renamed it to `testNFSDaemons()`. It's a very simple test, we should make it smarter soon though.

**Comment by leseb** [comment]
@galexrt can we re-trigger the CI? I believe the current error is not due to @travisn's PR.
Thanks.

**Comment by galexrt** [comment]
Merging for @travisn as he is getting on a plane. Good flight!

**Pull request merged**